src/osd/PG.cc

```diff
@@ -7689,6 +7689,15 @@ PG::RecoveryState::GetMissing::GetMissing(my_context ctx)
   context< RecoveryMachine >().log_enter(state_name);
 
   PG *pg = context< RecoveryMachine >().pg;
+  //find the first interval after peering completed
+  map<epoch_t, pg_interval_t>::const_reverse_iterator p = pg->past_intervals.rbegin();
+  for (; p != pg->past_intervals.rend(); ++p) {
+    if (p->second.last < pg->info.history.last_epoch_started) {
```

Comment:
history.last_epoch_started here isn't necessarily the most recent one.  To get that, you'd have to take the max over info.history.last_epoch_started for all osds in the PriorSet.